CopilotKit/packages/backend/src/utils/stream.ts

```diff
@@ -145,55 +149,61 @@ export function copilotkitStreamInterceptor(
             writeChatCompletionEnd(controller);
             await cleanup(controller);
             return;
-          } else if (debug) {
+          } // done == true (terminal case)
+
+          if (debug) {
             console.log("data: " + JSON.stringify(value) + "\n\n");
           }
 
-          let mode: "function" | "message" = value.choices[0].delta.tool_calls
-            ? "function"
-            : "message";
+          type Mode = { type: "function"; toolCall: ToolCallPayload } | { type: "message" };
```

Comment:
store mode type + associated value (toolCall) together.
Below, if we know we are in the type function, we also have a non-optional toolCall available, without having to repeat `value.choices[0].delta.tool_calls?.[0]` over and over again (which is esepcially error prone due to `?`, `!`, etc.)